src/Api/Users.php

```diff
@@ -553,6 +553,62 @@ public function removeImpersonationToken(int $user_id, int $impersonation_token_
         return $this->delete('users/'.self::encodePath($user_id).'/impersonation_tokens/'.self::encodePath($impersonation_token_id));
     }
 
+    /**
+     * @param int   $user_id
+     * @param array $params
+     *
+     * @return mixed
+     */
+    public function userPersonalAccessTokens(int $user_id, array $params = [])
+    {
+        $resolver = $this->createOptionsResolver();
+
+        $resolver->setDefined('state')
+            ->setAllowedValues('state', ['all', 'active', 'inactive'])
+        ;
+
+        return $this->get('users/'.self::encodePath($user_id).'/personal_access_tokens', $resolver->resolve($params));
+    }
+
+    /**
+     * @param int $user_id
+     * @param int $personal_access_token_id
+     *
+     * @return mixed
+     */
+    public function userPersonalAccessToken(int $user_id, int $personal_access_token_id)
+    {
+        return $this->get('users/'.self::encodePath($user_id).'/personal_access_tokens/'.self::encodePath($personal_access_token_id));
+    }
+
+    /**
+     * @param int         $user_id
+     * @param string      $name
+     * @param array       $scopes
+     * @param string|null $expires_at
+     *
+     * @return mixed
+     */
+    public function createPersonalAccessToken(int $user_id, string $name, array $scopes, ?string $expires_at = null)
```

Comment:
should expires at really be a string? how does a user know what to put here?

Reply:
We almost certainly want to take a DateTimeImmutable here, and probably to use a params array, like in `Groups::issues`.